File: app/src/main/java/ua/com/sober/getwords/mvp/models/ocr/OcrApiResponse.java

package ua.com.sober.getwords.mvp.models.ocr;

import java.util.ArrayList;
import java.util.List;
import javax.annotation.Generated;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import org.apache.commons.lang3.builder.ToStringBuilder;

@Generated("org.jsonschema2pojo")
public class OcrApiResponse {

    @SerializedName("ParsedResults")
    @Expose
    private List<ParsedResult> parsedResults = new ArrayList<ParsedResult>();
    @SerializedName("OCRExitCode")
    @Expose
    private Integer oCRExitCode;
    @SerializedName("IsErroredOnProcessing")
    @Expose
    private Boolean isErroredOnProcessing;
    @SerializedName("ErrorMessage")
    @Expose
    private String errorMessage;
    @SerializedName("ErrorDetails")
    @Expose
    private String errorDetails;
    @SerializedName("ProcessingTimeInMilliseconds")
    @Expose
    private String processingTimeInMilliseconds;

    /**
     * 
     * @return
     *     The parsedResults
     */
    public List<ParsedResult> getParsedResults() {
        return parsedResults;
    }

    /**
     * 
     * @param parsedResults
     *     The ParsedResults
     */
    public void setParsedResults(List<ParsedResult> parsedResults) {
        this.parsedResults = parsedResults;
    }

    /**
     * 
     * @return
     *     The oCRExitCode
     */
    public Integer getOCRExitCode() {
        return oCRExitCode;
    }

    /**
     * 
     * @param oCRExitCode
     *     The OCRExitCode
     */
    public void setOCRExitCode(Integer oCRExitCode) {
        this.oCRExitCode = oCRExitCode;
    }

    /**
     * 
     * @return
     *     The isErroredOnProcessing
     */
    public Boolean getIsErroredOnProcessing() {
        return isErroredOnProcessing;
    }

    /**
     * 
     * @param isErroredOnProcessing
     *     The IsErroredOnProcessing
     */
    public void setIsErroredOnProcessing(Boolean isErroredOnProcessing) {
        this.isErroredOnProcessing = isErroredOnProcessing;
    }

    /**
     * 
     * @return
     *     The errorMessage
     */
    public String getErrorMessage() {
        return errorMessage;
    }

    /**
     * 
     * @param errorMessage
     *     The ErrorMessage
     */
    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    /**
     * 
     * @return
     *     The errorDetails
     */
    public String getErrorDetails() {
        return errorDetails;
    }

    /**
     * 
     * @param errorDetails
     *     The ErrorDetails
     */
    public void setErrorDetails(String errorDetails) {
        this.errorDetails = errorDetails;
    }

    /**
     * 
     * @return
     *     The processingTimeInMilliseconds
     */
    public String getProcessingTimeInMilliseconds() {
        return processingTimeInMilliseconds;
    }

    /**
     * 
     * @param processingTimeInMilliseconds
     *     The ProcessingTimeInMilliseconds
     */
    public void setProcessingTimeInMilliseconds(String processingTimeInMilliseconds) {
        this.processingTimeInMilliseconds = processingTimeInMilliseconds;
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this);
    }

}

File: app/src/main/java/ua/com/sober/getwords/mvp/models/ocr/OcrApiResponse.kt
package ua.com.sober.getwords.mvp.models.ocr

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class OcrApiResponse {
    @SerializedName("ParsedResults")
    @Expose
    var parsedResults: List<ParsedResult>? = null
    @SerializedName("OCRExitCode")
    @Expose
    var ocrExitCode: Int = 0
    @SerializedName("IsErroredOnProcessing")
    @Expose
    var isErroredOnProcessing: Boolean = false
    @SerializedName("ErrorMessage")
    @Expose
    var errorMessage: String? = null
    @SerializedName("ErrorDetails")
    @Expose
    var errorDetails: String? = null
    @SerializedName("ProcessingTimeInMilliseconds")
    @Expose
    var processingTimeInMilliseconds: String? = null

    class ParsedResult {
        @SerializedName("TextOverlay")
        @Expose
        var textOverlay: TextOverlay? = null
        @SerializedName("FileParseExitCode")
        @Expose
        var fileParseExitCode: Int? = null
        @SerializedName("ParsedText")
        @Expose
        var parsedText: String? = null
        @SerializedName("ErrorMessage")
        @Expose
        var errorMessage: String? = null
        @SerializedName("ErrorDetails")
        @Expose
        var errorDetails: String? = null

        override fun toString(): String {
            return "ParsedResult(textOverlay=$textOverlay, fileParseExitCode=$fileParseExitCode, parsedText=$parsedText, errorMessage=$errorMessage, errorDetails=$errorDetails)"
        }
    }

    class TextOverlay {
        @SerializedName("Lines")
        @Expose
        var lines: List<Line>? = null
        @SerializedName("HasOverlay")
        @Expose
        var hasOverlay: Boolean? = null
        @SerializedName("Message")
        @Expose
        var message: String? = null

        override fun toString(): String {
            return "TextOverlay(lines=$lines, hasOverlay=$hasOverlay, message=$message)"
        }
    }

    class Line {
        @SerializedName("Words")
        @Expose
        var words: List<Word>? = null
        @SerializedName("MaxHeight")
        @Expose
        var maxHeight: Int? = null
        @SerializedName("MinTop")
        @Expose
        var minTop: Int? = null

        override fun toString(): String {
            return "Line(words=$words, maxHeight=$maxHeight, minTop=$minTop)"
        }
    }

    class Word {
        @SerializedName("WordText")
        @Expose
        var wordText: String? = null
        @SerializedName("Left")
        @Expose
        var left: Int? = null
        @SerializedName("Top")
        @Expose
        var top: Int? = null
        @SerializedName("Height")
        @Expose
        var height: Int? = null
        @SerializedName("Width")
        @Expose
        var width: Int? = null

        override fun toString(): String {
            return "Word(wordText=$wordText, left=$left, top=$top, height=$height, width=$width)"
        }
    }

    override fun toString(): String {
        return "OcrApiResponse(parsedResults=$parsedResults, ocrExitCode=$ocrExitCode, isErroredOnProcessing=$isErroredOnProcessing, errorMessage=$errorMessage, errorDetails=$errorDetails, processingTimeInMilliseconds=$processingTimeInMilliseconds)"
    }
}
